src/librustc/ty/context.rs

```diff
@@ -1997,3 +1999,13 @@ impl<T, R, E> InternIteratorElement<T, R> for Result<T, E> {
         Ok(f(&iter.collect::<Result<AccumulateVec<[_; 8]>, _>>()?))
     }
 }
+
+fn in_scope_traits<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, id: DefId)
+    -> Option<Rc<Vec<TraitCandidate>>>
+{
```

Comment:
I don't like the proliferation of `provide` methods, but I think making `trait_map` public in all of `rustc::ty` would be worse.